homeassistant/components/renault/sensor.py

```diff
@@ -0,0 +1,196 @@
+ATTR_PLUGGED = "plugged"
+ATTR_PLUG_STATUS = "plug_status"
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Old way of setting up platforms."""
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Renault entities from config entry."""
+    proxy = hass.data[DOMAIN][config_entry.unique_id]
+    entities = await get_entities(hass, proxy)
+    proxy.entities.extend(entities)
+    async_add_entities(entities, True)
+
+
+async def get_entities(hass, proxy: PyzeProxy):
+    """Create Renault entities for all vehicles."""
+    entities = []
+    for vehicle_link in proxy.get_vehicle_links():
+        vehicle_proxy = await proxy.get_vehicle_proxy(vehicle_link)
+        entities.extend(await get_vehicle_entities(hass, vehicle_proxy))
+    return entities
+
+
+async def get_vehicle_entities(hass, vehicle_proxy: PyzeVehicleProxy):
+    """Create Renault entities for single vehicle."""
+    entities = []
+    entities.append(RenaultBatteryLevelSensor(vehicle_proxy, "Battery Level"))
+    entities.append(RenaultChargeStateSensor(vehicle_proxy, "Charge State"))
+    entities.append(RenaultChargingPowerSensor(vehicle_proxy, "Charging Power"))
+    entities.append(RenaultPlugStateSensor(vehicle_proxy, "Plug State"))
+    entities.append(RenaultRangeSensor(vehicle_proxy, "Range"))
+    return entities
+
+
+class RenaultBatteryLevelSensor(RenaultBatteryDataEntity):
+    """Battery Level sensor."""
+
+    @property
+    def state(self):
+        """Return the state of this entity."""
+        data = self.coordinator.data
+        if "batteryLevel" in data:
+            return data.get("batteryLevel")
+        LOGGER.warning("batteryLevel not available in coordinator data %s", data)
+
+    @property
+    def device_class(self):
+        """Return the class of this entity."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return PERCENTAGE
+
+    @property
+    def icon(self):
+        """Icon handling."""
+        data = self.coordinator.data
+        chargestate = data["chargingStatus"] == 1
+        return icon_for_battery_level(battery_level=self.state, charging=chargestate)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of this entity."""
+        attrs = {}
+        attrs.update(super().device_state_attributes)
+        data = self.coordinator.data
+        if "batteryAvailableEnergy" in data:
+            attrs[ATTR_BATTERY_AVAILABLE_ENERGY] = data["batteryAvailableEnergy"]
+        if "batteryTemperature" in data:
+            attrs[ATTR_BATTERY_TEMPERATURE] = data["batteryTemperature"]
+        return attrs
+
+
+class RenaultChargingRemainingTimeSensor(RenaultBatteryDataEntity):
+    """Charging Remaining Time sensor."""
+
```

Comment:
device_class == DEVICE_CLASS_TIMESTAMP ?

But then I don't think this should be the remaining time, but the estimated time until full charge.